docs/api/ros2.md

```diff
@@ -49,26 +49,39 @@ For more details on the content of these topics, please have a look at their cor
  - [Odometry](odometry.md)
  - [User Interface](ui.md)
 
+If you have trouble seeing the topics using `ros2 topic list`, ensure that the robot's `RMW_IMPLEMENTATION` matches the one on your machine; see [Network Configuration](../setup/xml-config.md) for more information about <ins>R</ins>OS <ins>m</ins>iddle<ins>w</ins>are (RMW).
+Additionally, the command line `ros2 topic` utility can cache old information; try running it with additional arguments `ros2 topic list --no-daemon --spin-time 10`
```

Comment:
This is not 100% correct, but we could consider it a reasonable simplification.

The first `ros2 ...` command that you run will automatically create a `_ros2_daemon` process.
This process will cache discovery information (i.e. the location of remote endpoints) and future ROS processes will automatically get discovery information from it rather than actually performing a remote participant discovery step (I don't know how the daemon and the ros process communicate).